packages/accounts/src/services/accounts.ts

```diff
@@ -959,4 +933,280 @@ export class AccountsService implements AccountsServiceInterface {
     }
     return trx
   }
+
+  public async extendTrustline({
+    accountId,
+    amount,
+    autoApply
+  }: ExtendTrustlineOptions): Promise<void | TrustlineError> {
+    return this.adjustTrustline({
+      accountId,
+      amount,
+      mode: autoApply
+        ? AdjustTrustlineMode.AutoApply
+        : AdjustTrustlineMode.Extend
+    })
+  }
+
+  public async utilizeTrustline({
+    accountId,
+    amount
+  }: TrustlineOptions): Promise<void | TrustlineError> {
+    return this.adjustTrustline({
+      accountId,
+      amount,
+      mode: AdjustTrustlineMode.Utilize
+    })
+  }
+
+  public async revokeTrustline({
+    accountId,
+    amount
+  }: TrustlineOptions): Promise<void | TrustlineError> {
+    return this.adjustTrustline({
+      accountId,
+      amount,
+      mode: AdjustTrustlineMode.Revoke
+    })
+  }
+
+  public async settleTrustline({
+    accountId,
+    amount,
+    revolve
+  }: SettleTrustlineOptions): Promise<void | TrustlineError> {
+    return this.adjustTrustline({
+      accountId,
+      amount,
+      mode:
+        revolve === false
+          ? AdjustTrustlineMode.Settle
+          : AdjustTrustlineMode.Revolve
+    })
+  }
+
+  private async adjustTrustline({
+    accountId,
+    amount,
+    mode
+  }: {
+    accountId: string
+    amount: bigint
+    mode: AdjustTrustlineMode
+  }): Promise<void | TrustlineError> {
+    try {
+      const transfers: BalanceTransfer[] = []
+      const err = await transaction(
```

Comment:
Will this function behave if it is run for overlapping account trees in parallel?

I think [repeatable read](https://www.postgresql.org/docs/9.5/transaction-iso.html) + `SELECT FOR UPDATE` would prevent parallel modifications from trampling each other.

Reply:
Why wouldn't `SELECT FOR UPDATE` with the default [Read Committed](https://www.postgresql.org/docs/current/transaction-iso.html#XACT-READ-COMMITTED) suffice?

>UPDATE, DELETE, SELECT FOR UPDATE, and SELECT FOR SHARE commands behave the same as SELECT in terms of searching for target rows: they will only find target rows that were committed as of the command start time. However, such a target row might have already been updated (or deleted or locked) by another concurrent transaction by the time it is found. In this case, the would-be updater will wait for the first updating transaction to commit or roll back (if it is still in progress). If the first updater rolls back, then its effects are negated and the second updater can proceed with updating the originally found row. If the first updater commits, the second updater will ignore the row if the first updater deleted it, otherwise it will attempt to apply its operation to the updated version of the row. The search condition of the command (the WHERE clause) is re-evaluated to see if the updated version of the row still matches the search condition. If so, the second updater proceeds with its operation using the updated version of the row. In the case of SELECT FOR UPDATE and SELECT FOR SHARE, this means it is the updated version of the row that is locked and returned to the client.

Reply:
You're right, that should be fine. :+1: